modules/runtime-common/src/main/scala/com/snowplowanalytics/snowplow/runtime/Webhook.scala

```diff
@@ -42,32 +47,61 @@ object Webhook {
     }
   }
 
-  private implicit def logger[F[_]: Sync]: Logger[F] = Slf4jLogger.getLogger[F]
+  def resource[F[_]: Async, SetupAlert: Show: Eq](
+    config: Config,
+    appInfo: AppInfo,
+    httpClient: Client[F],
+    appHealth: AppHealth[F, SetupAlert, ?]
+  ): Resource[F, Unit] =
+    stream(config, appInfo, httpClient, appHealth).compile.drain.background.void
 
-  def create[F[_]: Async, Alert: Show](
-    config: Option[Config],
+  def stream[F[_]: Async, SetupAlert: Show: Eq](
+    config: Config,
     appInfo: AppInfo,
-    httpClient: Client[F]
-  ): Webhook[F, Alert] = new Webhook[F, Alert] {
-
-    override def alert(message: Alert): F[Unit] =
-      config match {
-        case Some(webhookConfig) =>
-          val request = buildHttpRequest[F, Alert](webhookConfig, appInfo, message)
-          Logger[F].info(show"Sending alert to ${webhookConfig.endpoint} with details of the setup error...") *>
-            executeHttpRequest[F](webhookConfig, httpClient, request)
-        case None =>
-          Logger[F].debug(s"Webhook monitoring is not configured, skipping alert: $message")
+    httpClient: Client[F],
+    appHealth: AppHealth[F, SetupAlert, ?]
+  ): Stream[F, Nothing] =
+    appHealth.setupHealth.discrete.changes
```

Comment:
I know what you mean.  Currently we are accessing the class field in three places:
- Here in the webhook stream
- In the HealthProbe
- In the AppHealth spec

So the change you suggest makes sense if all of those places switched to use a method instead of a class field.  But I don't think that's possible to switch all of them -- unless you see a way?  Unless we can switch all of them, then I'm ok with calling the class field in the Webhook class.